### PR TITLE
Use cluster features for ASYNC ESQL tests

### PR DESCRIPTION
**Pull request by ldematte**
With https://github.com/elastic/elasticsearch/pull/104279 we introduced a way to perform feature checks against the "original" (or "old") cluster in upgrade and mixed cluster tests.
With this PR we add a feature for ASYNC and we replace the version check with a feature check.

### CODE REVIEW

**Comment by ldematte** [comment]
@elasticmachine rerun elasticsearch-ci/part-4

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java [hunk 1]
```diff
@@ -12,10 +12,19 @@
 import org.elasticsearch.features.NodeFeature;
 
 import java.util.Map;
+import java.util.Set;
 
 public class EsqlFeatures implements FeatureSpecification {
+
+    public static final NodeFeature ASYNC_API_SUPPORTED = new NodeFeature("esql.async_api_supported");
```
Comment (ChrisHegarty) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java:19]: could we please call this "ASYNC_QUERY_API.xx" ?

**Review by ChrisHegarty — COMMENTED** [review]
ah ha! this is cool. thanks @ldematte

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java (hunk 1 above)
Comment (ldematte) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java:19]: No problem with changing it to anything you like but... what should be "xx"?
Some area of the API/some action?

Comment (ldematte) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java:19]: @ChrisHegarty gentle reminder to give your feedback here :)

**Review by mark-vieira — APPROVED** [review]

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java (hunk 1 above)
Comment (ChrisHegarty) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java:46]: `ASYNC_QUERY` is a good name.

**Review by ChrisHegarty — APPROVED** [review]
LGTM

**Pull request merged**